src/main/scala/se/nimsa/sbx/app/routing/SeriesTagsRoutes.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2014 Lars Edenbrandt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package se.nimsa.sbx.app.routing
+
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.pattern.ask
+import se.nimsa.sbx.app.SliceboxBase
+import se.nimsa.sbx.metadata.MetaDataProtocol._
+import akka.http.scaladsl.model.StatusCodes.{BadRequest, Created, InternalServerError, NoContent, NotFound}
+
+import scala.util.{Failure, Success}
+
+
+// POST /metadata/series/{id}/seriestags
+// DEL  /metadata/series/{seriesId}/seriestags/{seriesTagId}
+
+// GET  /metadata/seriestags
+
+
+
+// TODO:
+// Rename tag
+// Remove tag (completely, not just remove tag from series as is possible today)
+// Do not remove unused tags now that there is a way to remove them manually
+
+
+trait SeriesTagsRoutes {
+  this: SliceboxBase =>
+
+  def seriesTagsRoutes: Route =
+    pathPrefix("metadata") {
+      pathPrefix("seriestags") {
+        pathEndOrSingleSlash {
+          get {
+            onSuccess(metaDataService.ask(GetSeriesTags)) {
+              case SeriesTags(seriesTags) =>
+                complete(seriesTags)
+            }
+
+          } ~ post {
+            entity(as[SeriesTag]) { seriesTag =>
+              onComplete(metaDataService.ask(CreateSeriesTag(seriesTag))) {
+                case Success(tag: SeriesTag) => complete((Created, tag))
+                case Failure(ex) => {
+                  val msg = ex.getMessage
+                  if (msg.contains("idx_unique_series_tag_name")) {
```

Comment:
This seems a little too "stringly typed". Should we introduce a new Exception? Or throw an IllegalArgumentException which is always mapped to BadRequest.

Reply:
found no better solution, exception is jdbc driver specific